packages/shared/src/hooks/useReferralCampaign.ts

```diff
@@ -36,17 +36,14 @@ export const campaignToReferralTargetCountMap: Record<
 > = {};
 
 const campaignFeatureFlagMap: Partial<
-  Record<ReferralCampaignKey, Features<string>>
+  Record<ReferralCampaignKey, Feature<string>>
 > = {};
 
 const useReferralCampaign = ({
   campaignKey,
 }: UseReferralCampaignProps): UseReferralCampaign => {
-  const { flags } = useContext(FeaturesContext);
   const featureFlag = campaignFeatureFlagMap[campaignKey];
-  const isCampaignEnabled = featureFlag
-    ? !!getFeatureValue(featureFlag, flags)
-    : true;
+  const isCampaignEnabled = useFeatureIsOn(featureFlag);
```

Comment:
Haven't we concluded this already? Do we need it?

Reply:
Yes, though I believe this was built to be dynamic, hence, the featureFlag is coming from a map. cc @capJavert

Reply:
Yes, if campaign is behind feature flag it will only enable hook when campaign is active.